test/common/concurrent_lru_cache_common.h

```diff
@@ -114,10 +115,17 @@ namespace concurrent_lru_cache_helpers {
         }
 
         ~instance_counter() {
-            if (! --(*my_p_count))
+            bool is_last = ! --(*my_p_count);
+#if __GNUC__ == 12
+        // GCC 12 warns about using my_p_count after delete.
+        // The test was investigated and no problems were detected
+        // The following statement silence the warning
+        static bool unused_is_last = is_last;
+        utils::suppress_unused_warning(unused_is_last);
```

Comment:
Possible writing from multiple threads to a static variable is not UB?
Or is the ``~instance_counter`` call synchronized?

Reply:
This static variable will be written only once on initialization and static variable initialization is thread safe

Reply:
@pavelkumbrasev 
You're right! I did not notice that this is an initialization.